Develop

warenga opened the pull request:
#### What does this PR do?
- Adds mocha tests for user, document and role routes
- Adds user and document pagination routes
- Adds user and document search routes
- Eslints all files

 #### Description of Task to be completed?
- Create test suites for all routes
- Implement the following routes:
   - GET /users/?limit={integer}&offset={integer}
   - GET /documents/?limit={integer}&offset={integer}
   - GET /search/users/?q={username}
   - GET /search/documents/?q={doctitle}
- Include an eslint file that extends the airbnb standard in the root directory to lint server side code

 #### How should this be manually tested?
- Clone the repo
- run `npm install`
- run `npm run start:dev`
- to run tests
   - run `npm test`
- to run pagination routes:
   - Copy/Paste the above routes and replace the {integer} with a number where:
      limit: the number of records you want to query 
      offset: How many instances should be skipped
- to run search routes:
    - for users: replace {username} with the name of the user you want to search
    - for documents: replace {title} with the title of the doc you are searching for

 #### What are the relevant pivotal tracker stories?

- 140472095
- 141777673
- 141880597
- 141881983

Commit:
Revert "#140472073 user and role endpoints"

Commit:
Merge pull request #7 from andela-wmaina/revert-6-feature/140472073/user-and-role-endpoints

140472073 user and role endpoints

Commit:
(feature): Add tests for user, documents and roles endpoints

140472095

Commit:
chore(npm scripts): modify npm script

- add npm start:dev to run development server
- add npm migrate to run migrations
- add npm seed to run migrations and seeders

140472095

Commit:
feature(seeders): add seeders to populate test db

- add document seeders
- add user seeders
- add roles seeders

140472095

Commit:
feature(test spec): Add test specs

- add document.spec tests to test document routes
- add user.spec tests to test user routes
- add role.spec tests to test role routes
- add auth.spec tests to test user creation and login

140472095

Commit:
chore(npm seed): Add npm run seed in test script

140472095

Commit:
feature(routes): Add pagination routes to user and documents

141777673

Commit:
feature(pagination): Add user and document pagination methods

- Also change user functions to methods in a class

141777673

Commit:
chore(class): Change role functions to methods in a class

140472095

Commit:
chore(routes): Change routes position

This change is so that certain routes fall under the middleware specified in User.Router

140472095

Commit:
feature(model): Add access variable to document model

140472095

Commit:
fix(pagination): refactor pagination

Made use of query parameters

141777673

Commit:
chore(travis): add travis.yml file

141777673

Commit:
feature(search routes): Add user and documents search routes

141880597

Commit:
feature(search routes): Add test suites to test search routes

141880597

Commit:
chore(Eslint): ensure all files are linted

141881983

Commit:
Merge branch 'feature/140472095/backend-tests' into develop

Commit:
Merge branch 'develop' of https://github.com/andela-wmaina/document-management into develop

Commit:
Merge pull request #9 from andela-wmaina/feature/141777673/add-pagination-routes

141777673 Add pagination routes

Commit:
Merge branch 'master' into feature/141880597/add-search-routes

Commit:
Merge branch 'develop' into chore/141881983/eslint-files

Commit:
Merge branch 'develop' into feature/141880597/add-search-routes

Commit:
Merge pull request #10 from andela-wmaina/feature/141880597/add-search-routes

#141880597 Add search routes

Commit:
Merge branch 'develop' into chore/141881983/eslint-files

Commit:
fix(environment variable): Fix issues arising with dotenv file

141881983

Commit:
Merge branch 'chore/141881983/eslint-files' of https://github.com/andela-wmaina/document-management into chore/141881983/eslint-files

Commit:
fix(chore): add index.js controller file

141881983

Commit:
fix(dotenv): fix environment variables bug

- Add postgres dialect to test key in config.json

141881983

Commit:
chore(travis): add postgres to travis file

141881983

Commit:
Merge pull request #11 from andela-wmaina/chore/141881983/eslint-files

#141881983 Eslint files

nyambati commented on .env:
```diff
@@ -0,0 +1,3 @@
+DATABASE_URL=postgres://vopnksci:CFmTZ3i25saUxDNmymi5-Rt_z4FDR_hH@babar.elephantsql.com:5432/vopnksci
```
🔴 Thou shall not version this file

nyambati commented on server.js:
```diff
@@ -3,7 +3,7 @@ const app = express();
 const bodyParser = require('body-parser');
 const routes = require('./server/routes')
-
+const dotenv = require('dotenv').config()
```
Wrap this in if statement

nyambati commented on server/controllers/document.js:
```diff
@@ -1,19 +1,42 @@
+      })
+      .then((document) => {
+        res.status(201).send({
+          message: 'Successful entry',
```
Provide descriptive messages back to the user

nyambati commented on server/controllers/document.js:
```diff
@@ -1,19 +1,42 @@
+        .then((document) => {
+          if (!document) {
+            return res.status(404).send({
+              message: 'No Users'
```
How did users get here?????? ❓

nyambati commented on server/controllers/document.js:
```diff
@@ -36,51 +59,81 @@ class DocumentController {
     return Document
-      .findById(req.params.documentId)
-      .then(document => {
+      .findById(req.params.id)
```
Don't find just update

nyambati commented on server/controllers/document.js:
```diff
@@ -36,51 +59,81 @@ class DocumentController {
+      });
+  }
+
+  findByName(req, res) {
```
Use one find method by making use of or

Commit:
chore(dotenv): requires dotenv only on development enviromnent

142211463

Commit:
chore(message): Refactor response message to users

- Change test suites to reflect the changed response message
- Also queries document list where access is public
- Gives a document's access property a default value of 'private' if none is specified

142211463

Commit:
chore(update): Refactors update method

- Removes find before update in User and Document update method

142211463

Commit:
chore(tests): Adds a test under POST Documents

- Adds a test to check when a token is not provided when creating a document

142211463

Commit:
fix(bcrypt): fix travis.yml bcrypt bug

- Change bcrypt package to bcrypt-nodejs

142211463

Commit:
fix(logs on test): Sets logging to false on test environment

142211463

Commit:
fix(logging): disables raw sql logs

- Removes logging from config.json and places it when a new Sequelize instance is created.

142211463

Commit:
fix(tests): Passes previously failing tests

142211463

Commit:
fix(console.log): removes console.log from test suite

142211463

Commit:
chore(coveralls): sets up coveralls

- Also adds a coveralls badge

142254839

Commit:
chore(coveralls): adds coveralls configuration to travis.yml

142254839

Commit:
fix(coveralls): fix coveralls error on travis

142254839

Commit:
fix[coveralls]: Installs coveralls and istanbul before_script in travis

142254839

Commit:
fix[coveralls]: fix istanbul bug

142254839

Commit:
Merge pull request #13 from andela-wmaina/chore/142254839/integrate-coveralls

142254839 Integrate coveralls

Commit:
chore(coverage): Adds coverage folder to gitignore file

142211463

Commit:
feature(middleware): Adds middleware to check user permission

- Adds error handling functionality

142211463

Commit:
chore(tests): Adds more tests to test error handling

142211463

Commit:
fix(remove yarn): Uninstall yarn to fix travis bug

142211463

Commit:
fix(bcrypt): Removes bcrypt from package.json

- To fix travis error

142211463

Commit:
chore(front-end config): Sets up front end configurations

- sets up webpack to compile and run react
- sets up express server to use webpack
- installs necessary dev dependencies for react
- adds eslint rules for jsx and react

140472131

Commit:
Merge pull request #14 from andela-wmaina/chore/142211463/changes-from-code-review

142211463 Checkpoint review feedback

Commit:
chore(remove .env): Removes .env file from git repo

Commit:
Merge branch 'develop' of https://github.com/andela-wmaina/document-management into develop

Commit:
feature(document components): Adds components to render documents api

- Configures the store
- Adds index routes and docs routes
- Fetches and displays all documents from the document get request
- Adds a button for adding a new document
- Runs a post request when button is clicked
- Adds MaterialUI elements to the UI

142547147

Commit:
chore(API documentation): Adds API documentation

- A .md file that contains the document management API blueprint

142673869

Commit:
fix(travis): fix document test fail

142673869

Commit:
feature(error handling): Checks if user input is valid

142750017

Commit:
feature(documents): Ensures a user sees their private and all public

142750017

Commit:
fix(.env): remove .env file

140472131

Commit:
fix(unnecesary code): Removes unnecessary code

142673869

Commit:
fix(travis bug): Adds Port number to server.js

140472131

Commit:
fix(.env): remove .env file

142547147

Commit:
fix(travis bug): Add port number to server.js

142673869

Commit:
Merge pull request #16 from andela-wmaina/feature/142547147/document-component

142547147 Document component

Commit:
Merge branch 'chore/140472131/front-end-configuration' into chore/142673869/api-documentation

Commit:
fix(unnecessary files): Removes unnecessary files

142673869

Commit:
Merge branch 'chore/142673869/api-documentation' of https://github.com/andela-wmaina/document-management into chore/142673869/api-documentation

Commit:
fix(unnecessary files): removes unnecessary files

142673869

Commit:
Merge pull request #17 from andela-wmaina/chore/142673869/api-documentation

142673869 API documentation

Commit:
Revert "Merge branch 'chore/142673869/api-documentation' of https://github.com/andela-wmaina/document-management into chore/142673869/api-documentation"

This reverts commit b94da1299d9d6b9672548f367137eca8629eb6d4, reversing
changes made to 7a307f78e68feb29f181804e35e4af645852dd03.

Commit:
chore(documentation): Sets documentation to root url

- Creates a documentation html file

142673869

Commit:
Merge pull request #19 from andela-wmaina/chore/142673869/create-api-documentation

142673869 Sets documentation to root url

Commit:
chore(jsdoc): Adds jsdoc and cleans code

142750017

Commit:
Merge branch 'develop' into feature/142750017/second-review-feedback

Commit:
fix(travis bug): remove duplicate app declaration

142750017

Commit:
fix(travis bug): remove duplicate port declaration

142750017

Commit:
fix(travis bug): globalize sequelize

142750017

Commit:
fix(travis bug): fix test

142750017

Commit:
fix(travis bug): fix server

142750017

Commit:
fix(Object.values bug): Replaces object.values with Object.keys

142750017

Commit:
fix(role model): remove required as a field in role model

142750017

Commit:
fix(travis test): Returns required in role model

142750017

Commit:
fix(travis): Refactor roles model

142750017

Commit:
Merge pull request #20 from andela-wmaina/feature/142750017/second-review-feedback

142750017 Second Review Feedback

Commit:
fix(user response): Removes unnecessary information from user response

142834489

Commit:
feature(user response): Filters returned user object

- Removes unnecessary output

142834489

Commit:
feature(route): Adds /users/:id/documents endpoint

 142834489

Commit:
chore(response status): Corrects misleading response status

142834489

Commit:
chore(filter method): Moves filter method to helpers folder

142834489

Commit:
chore(setup): Configure front-end environment

- Install react-redux dependencies
- Install Material-UI
- Set up webpack.config file
- Set up webpack in server.js

140472131

Commit:
feature(index.html): Adds client folder and index.html file

140472131

Commit:
feature(homepage): Sets up the home page and naviagtion bar

- configures routes for home and document

14254714

Commit:
feature(routes): adds routes file that call home page

142547147

Commit:
feature(document): adds document components

- Adds component to list all documents
- Adds component to add a new document
- Adds component to edit a new document

142547147

Commit:
feature(document routes): combines private and public docs

 142547147

Commit:
feature(edit document): enables document to be updated

142547147

Commit:
feature(document): adds document delete functionality

142547147

Commit:
feature(auth): adds authentication components

- adds signup component and form

142547147

Commit:
feature(login component): adds login component

- adds login functionality

142547147

Commit:
feature(logout): adds logout functionality

142547147

Commit:
chore(): remove unnecessary code

- Also replaces "user id" placeholder with the actual user id

142547147

Commit:
feature(user profile): Adds a profile page for users

- Allows users to edit their own data in the profile page

143021331

Commit:
feature(logout): add logout routes to allow user logout

143021331

Commit:
Merge pull request #22 from andela-wmaina/feature/143021331/user-components

#143021331 user components

Commit:
feature(search): adds search functionality

- Enables a user to search documents

142547147

Commit:
Merge branch 'feature/142547147/documents-component' of https://github.com/andela-wmaina/document-management into feature/142547147/documents-component

Commit:
fix(document): removes horizontal scroll bar

142547147

Commit:
fix(syntax fix): clean up code

142547147

Commit:
Merge pull request #21 from andela-wmaina/feature/142547147/documents-component

#142547147 documents component

Commit:
fix(public): fix heroku error

140472131

Commit:
feature(profile pic): Add profic pic placeholder

140472131

Commit:
fix(heroku): move babel to dependencies in package.json

140472131

Commit:
fix(heroku): installs and saves babel-polyfill

140472131

Commit:
chore(search textbox): refactor hint message

140472131

Commit:
Merge pull request #23 from andela-wmaina/feature/142834489/backend-defense-feedback

142834489 backend defense feedback

Commit:
Revert "142834489 backend defense feedback"

Commit:
Merge pull request #24 from andela-wmaina/revert-23-feature/142834489/backend-defense-feedback

Revert "142834489 backend defense feedback"

Commit:
Merge pull request #25 from warenga/feature/143021331/user-components

Add User components

Commit:
Merge pull request #26 from warenga/feature/140472131/front-end

Feature/140472131/front end

Commit:
Hide password in password fields

Commit:
Update react version and react router

Commit:
Merge pull request #27 from warenga/landing_page

Update react version and react router

Commit:
Fix babel and signout issues

Commit:
Fix react router issues

Commit:
Fix signup issue

Commit:
Fix render documents

Commit:
Fix deleting bug
